packages/hint-doctype/tests/tests.ts

```diff
@@ -0,0 +1,238 @@
+import { getHintPath } from 'hint/dist/src/lib/utils/hint-helpers';
+import { HintTest } from '@hint/utils-tests-helpers/dist/src/hint-test-type';
+import * as hintRunner from '@hint/utils-tests-helpers/dist/src/hint-runner';
+
+const hintPath = getHintPath(__filename);
+
+const tests: HintTest[] = [
+    {
+        name: 'DOCTYPE is not in the first line should fail',
+        reports: [{ message: `DOCTYPE is not in the first line.` }],
+        serverConfig: {
+            '/': {
+                content: `<p><span></span>
+                <head></head>
+                <!DOCTYPE html>
+                `,
+                headers: { 'Content-Type': 'text/html' }
+            }
+        }
+    },
+    {
+        name: 'DOCTYPE found more than once should fail',
+        reports: [{ message: `There is more than one DOCTYPE in the document.` }],
+        serverConfig: {
+            '/': {
+                content: `<!DOCTYPE html>
+                <p><span></span>
+                <head></head>
+                <!DOCTYPE html>
+                `,
+                headers: { 'Content-Type': 'text/html' }
+            }
+        }
+    },
+    {
+        name: 'HTML with no content should fail',
+        reports: [{ message: `Resource has no content.` }],
+        serverConfig: {
+            '/': {
+                content: '',
+                headers: { 'Content-Type': 'text/html' }
+            }
+        }
+    },
+    {
+        name: 'DOCTYPE not found should fail',
+        reports: [{ message: `The resource does not contain a valid DOCTYPE (e.g. \`<!doctype html>\`).` }],
```

Comment:
One last thing: it doesn't look like any of the tests validate the provided `ProblemLocation`. It can be added as an optional `position` property on the report:

```ts
reports: [{ message '...', position: { line: 0, col: 0 }]
```

I'd be particularly interested in having some validation for the more complicated positions (e.g. the test `DOCTYPE appearing more than once should fail`).